src/drivers/SpiMaster.h

```diff
@@ -1,65 +1,70 @@
 #pragma once
 #include <cstddef>
 #include <cstdint>
-
-#include <FreeRTOS.h>
-#include <semphr.h>
-#include <task.h>
+#include <algorithm>
 
 namespace Pinetime {
   namespace Drivers {
-    class SpiMaster {
-    public:
-      enum class SpiModule : uint8_t { SPI0, SPI1 };
-      enum class BitOrder : uint8_t { Msb_Lsb, Lsb_Msb };
-      enum class Modes : uint8_t { Mode0, Mode1, Mode2, Mode3 };
-      enum class Frequencies : uint8_t { Freq8Mhz };
-      struct Parameters {
-        BitOrder bitOrder;
-        Modes mode;
-        Frequencies Frequency;
-        uint8_t pinSCK;
-        uint8_t pinMOSI;
-        uint8_t pinMISO;
+    template <typename T>
+    concept IsSpiMaster =
```

Comment:
(outsider suggestion) (ditto for other places) Concepts should be named like adjectives rather than predicates, i.e. `SpiMaster` rather than `IsSpiMaster`.  This convention is established in the standard ([`same_as`, `derived_from`, `destructible`, `regular`, ...](https://en.cppreference.com/w/cpp/concepts)) and is in line with how other languages treat traits and trait-like features

Reply:
I agree with this. However... this rule is not easy to apply here : 
 - in drivers/SpiMaster.h, there's `Pinetime::Drivers::IsSpiMaster`, the concept definition
 - in drivers/SpiMaster.h, there's also `Pinetime::Drivers::Interface::SpiMaster`, the actual driver, that owns an `impl` of type `IsSpiMaster`.
 - in port/infinitime.h, we declare `PineTime::Drivers::SpiMaster`, the type that will be used in the rest of the code.

Renaming the concept from `IsSpiMaster` to ` SpiMaster` will conflict with `SpiMaster` in InfiniTime.h. I'm open to suggestions regarding the naming of the types and namespaces, though!